File: radio-core/src/main/java/ua/hope/radio/core/RadioApplication.java
package ua.hope.radio.core;

import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;

/**
 * Created by vitalii on 1/11/18.
 */

public class RadioApplication extends Application {

	@Override
	protected void attachBaseContext(Context base) {
		super.attachBaseContext(base);
		MultiDex.install(this);
	}
}

File: app/src/main/kotlin/ua/hope/radio/RadioApplication.kt
package ua.hope.radio

import androidx.multidex.MultiDexApplication
import com.google.firebase.analytics.FirebaseAnalytics
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.core.context.startKoin
import org.koin.dsl.module
import timber.log.Timber
import ua.hope.radio.player.PlayerViewModel
import ua.hope.radio.utils.CrashlyticsCrashReportingTree

class RadioApplication : MultiDexApplication() {
    private val koinModule = module {
        viewModel { PlayerViewModel() }
    }

    override fun onCreate() {
        super.onCreate()

        configureTimber()
        startKoin{
            modules(koinModule)
        }
        FirebaseAnalytics.getInstance(this).setAnalyticsCollectionEnabled(!BuildConfig.DEBUG)
    }

    private fun configureTimber() {
        if (BuildConfig.BUILD_TYPE == "debug") {
            Timber.plant(Timber.DebugTree())
        } else {
            Timber.plant(CrashlyticsCrashReportingTree())
        }
    }
}
